src/librados/librados.cc

```diff
@@ -1266,6 +1266,26 @@ int librados::IoCtx::read(const std::string& oid, bufferlist& bl, size_t len, ui
   return io_ctx_impl->read(obj, bl, len, off);
 }
 
+int librados::IoCtx::repair_read(const std::string& oid, bufferlist& bl,
+				 size_t len, uint64_t off, int flags,
+				 int32_t osdid, epoch_t e)
+{
+  auto c = librados::Rados::aio_create_completion();
```

Comment:
```diff
@@ -1268,19 +1268,16 @@ int librados::IoCtx::read(const std::string& oid, bufferlist& bl, size_t len, ui

 int librados::IoCtx::repair_read(const std::string& oid, bufferlist& bl,
                                 size_t len, uint64_t off, int flags,
-                                int32_t osdid, epoch_t e)
+                                int32_t osdid, epoch_t epoch)
 {
-  auto c = librados::Rados::aio_create_completion();
-  int r = aio_repair_read(oid, c, &bl, len, off, 0, flags, osdid, e);
-  if (r < 0) {
-    c->release();
-    return r;
-  }
-  c->wait_for_complete();
-  r = c->get_return_value();
-  if (r >= 0)
-    io_ctx_impl->set_sync_op_version(c->get_version64());
-  c->release();
+  if (len > (size_t) INT_MAX)
+    return -EDOM;
+  ObjectReadOperation op;
+  io_ctx_impl->prepare_assert_ops(&op.impl->o);
+  op.read(off, len, &bl, nullptr);
+  // Only 1 flag can be passed from user
+  flags &= CEPH_OSD_OP_FLAG_FAILOK;
+  int r = operate_repair_read(oid, &op, &bl, osdid, epoch, flags);
   if (r < 0)
     return r;
   return bl.length();
```

i'd say it's more of a trade-off. probably, the call path looks more reasonable this way: 

before this change:
- repair_read() => aio_repair_read()
- aio_repair_read() => aio_operate_repair_read()

after this change:
- repair_read() => operate_repair_read() => aio_operate_repair_read(
- aio_repair_read() => aio_operate_repair_read()